src/Generators/Microsoft.Gen.Logging/Parsing/Parser.cs

```diff
@@ -639,67 +617,125 @@ private void CheckTagNamesAreUnique(LoggingMethod lm, Dictionary<LoggingMethodPa
+                ITypeSymbol? typeSymbol = ms switch
                 {
-                    var gm = p.GetMethod;
-                    if (gm != null)
-                    {
-                        if (ParserUtilities.IsBaseOrIdentity(gm.ReturnType, symbol, _compilation))
-                        {
-                            if (!sm.Compilation.IsSymbolAccessibleWithin(gm, originType, type))
-                            {
-                                continue;
-                            }
+                    IFieldSymbol fs => fs.Type,
+                    IPropertySymbol ps => ps.Type,
+                    _ => default,
+                };
 
-                            if (member == null)
-                            {
-                                member = p.Name;
-                                isNullable = gm.ReturnType.NullableAnnotation == NullableAnnotation.Annotated;
-                            }
-                            else
-                            {
-                                return (null, p, isNullable);
-                            }
-                        }
+                if (typeSymbol is null)
+                {
+                    continue;
+                }
+
+                if (!onMostDerivedType && ms.DeclaredAccessibility == Accessibility.Private)
+                {
+                    continue;
+                }
+
+                if (!ms.CanBeReferencedByName)
+                {
+                    continue;
+                }
+
+                if (ParserUtilities.IsBaseOrIdentity(typeSymbol, loggerSymbol, _compilation))
+                {
+                    if (loggerMember == null)
+                    {
+                        loggerMember = ms.Name;
+                        isNullable = typeSymbol.NullableAnnotation == NullableAnnotation.Annotated;
+                    }
+                    else if (!onMostDerivedType)
+                    {
+                        // we found a public logger member on a base class, prefer the existing one
+                        // because it's more likely to be the one the user intended
+                        // the dotnet logging generator doesn't support ILogger properties so this matches that behavior
+                        continue;
+                    }
+                    else
+                    {
+                        return (null, ms, isNullable);
                     }
                 }
-                else if (m is IFieldSymbol f)
+                else
+                {
+                    _ = shadowedNames.Add(ms.Name);
+                }
+            }
+
+            onMostDerivedType = false;
+            currentClassType = currentClassType.BaseType;
+        }
+
+        // We prioritize fields over primary constructor parameters and avoid warnings if both exist.
+        if (loggerMember is not null)
+        {
+            return (loggerMember, null, isNullable);
```

Comment:
[nit/readability] add parameter names for "opaque" arguments (i.e., `true/false/null/""`):
```suggestion
            return (loggerMember, secondMember: null, isNullable);
```

_In some cases_, this may be even better:
```suggestion
            return (member: loggerMember, secondMember: null, isNullable);
```



Ditto other places.